models/Document/PageSnippet.php

```diff
@@ -412,12 +412,11 @@ public function getEditable(string $name)
 
     /**
      * @param int|string|null $contentMasterDocumentId
-     *
      * @return $this
      *
      * @throws \Exception
      */
-    public function setContentMasterDocumentId($contentMasterDocumentId)
+    public function setContentMasterDocumentId($contentMasterDocumentId, bool $validate=false)
```

Comment:
BC Break! since this could be extended in custom document types e.g. `headlessdocuments`.

Reply:
Add a property instead for the checks then it is up to the custom implementations to adapt it.

Reply:
Or additionally you can use `func_get_args()` and comment the additional parameter `/* , bool $validate=false */`